Detectors/TRD/workflow/include/TRDWorkflow/NoiseCalibSpec.h

```diff
@@ -42,19 +44,67 @@ class TRDNoiseCalibSpec : public o2::framework::Task
     auto digits = pc.inputs().get<gsl::span<Digit>>("trddigits");
     auto trigRecs = pc.inputs().get<gsl::span<TriggerRecord>>("trdtriggerrec");
 
-    // TODO:
-    // Here we have the digits and trigger records available for a given time frame.
-    // We should extract the relevant information (ADC sum per pad and number of ADC values added for each pad)
-    // and store that in padAdcInfo which is then send to the aggregator were another device needs to be prepared
-    // which combines the information from all different EPNs and creates the CCDB object
+    // Obtain rough time from the data header (done only once)
+    if (mStartTime == 0) {
+      o2::dataformats::TFIDInfo ti;
+      o2::base::TFIDInfoHelper::fillTFIDInfo(pc, ti);
+      if (!ti.isDummy()) {
+        mStartTime = ti.creation;
+      }
+    }
 
-    PadAdcInfo padAdcInfo;
+    // process the digits, as long as we have not reached the limit
+    if (!mCalibrator.hasEnoughData()) {
+      mCalibrator.process(digits);
+    } else {
+      if (!mHaveSentOutput) {
+        LOGP(important, "Enough data received after {} TFs seen, finalizing noise calibration", mNTFsProcessed);
+        mCalibrator.collectChannelInfo();
+        sendOutput(pc.outputs());
+        mHaveSentOutput = true;
+      } else {
+        if ((mNTFsProcessed % 1000) == 0) {
+          LOGP(important, "Not processing anymore. Seen {} TFs in total. Run can be stopped", mNTFsProcessed);
+        }
+      }
```

Comment:
I am not sure what statistics we want to collect, and if we want to restrict ourselves to a single noise calib object with a fixed size per run. 
At the moment, we require 100 billion digits. With approx. 1 M channels, that's 100k events. That's plenty of statistics, and at several 100Hz readout rate, this should be done in less than 10min.
That's probably a good setting for running online - if a single EPN can handle this.

If we process a run offline, where we have limited statistics, how would this work? We might not have the high statistics on (local) disk.

Reply:
The 2 CCDB objects are not really necessary -  I just thought about some obsucre test scenarios. 
I overlooked the possibility to pass the config key. That resolves this issue.